openedx_ledger/views.py

```diff
@@ -1,6 +1,62 @@
 """
 Views for the openedx_ledger app.
 """
-# from django.shortcuts import render
+import logging
 
-# Create your views here.
+from django.http import HttpResponseRedirect
+from django.shortcuts import render
+from django.urls import reverse
+from django.views.generic import View
+
+from openedx_ledger.api import reverse_full_transaction
+from openedx_ledger.models import Transaction
+from openedx_ledger.signals.signals import ADMIN_INVOKED_TRANSACTION_UNENROLL
+
+logger = logging.getLogger(__name__)
+
+
+class ReverseTransactionView(View):
+    """
+    Admin view for the form responsible for reversing transaction objects and emitting an unenrollment signal.
+    """
+    template = "edx_ledger/admin/reverse_transaction.html"
+
+    def get(self, request, transaction_id):
+        """
+        Handle GET request - render "Reverse Transaction" form.
+
+        Arguments:
+            request (django.http.request.HttpRequest): Request instance
+            transaction_uuid (str): Enterprise Customer UUID
+
+        Returns:
+            django.http.response.HttpResponse: HttpResponse
+        """
+        return render(
+            request,
+            self.template,
+            {'transaction': Transaction.objects.get(uuid=transaction_id)}
+        )
+
+    def post(self, request, transaction_id):
+        """
+        Handle POST request - handle form submissions.
+        """
+        logger.info(
+            "Reversing transaction and sending admin invoked transaction unenroll signal for transaction: "
+            f"{transaction_id}"
+        )
+        transaction = Transaction.objects.get(uuid=transaction_id)
+        # TODO: determine how to generate idempotency key
+        reversal = reverse_full_transaction(
+            transaction,
+            idempotency_key=f"admin-invoked-reverse-{str(transaction.uuid)}"
+        )
+        ADMIN_INVOKED_TRANSACTION_UNENROLL.send(
```

Comment:
I'm actually now wondering if we can leverage the reversal signal instead of needing the admin. Is there ever a point where a reversal shouldn't be followed by an un-enrollment? What if we just registered a receiver for `TRANSACTION_REVERSED` in subsidy that made the call to unenroll on platform?

Reply:
right, you don't want `ADMIN_INVOKED_TRANSACTION_UNENROLL` to exist anymore.  But `reverse_full_transaction()` is going to send the `TRANSACTION_REVERSED` signal for you, you don't need to do it in this view.